I am working on the Census Demographic ACS dataset. 

Hypothesis: There will be a strong relationship between the demographic composition of a county and the spread of COVID-19. Specifically, counties with a higher population density, a larger percentage of elderly residents, or a larger percentage of minority populations may affect COVID-19 case counts and death rates.


Variable Dictionary:

Variable Name: GEO_ID

Description: The unique identifier for the geographic region (typically the county) based on FIPS codes.
Data Type: String

Variable Name: NAME

Description: The name of the geographic area, usually the county name followed by the state.
Data Type: String

Variable Name: DP05_0001E (Total Population Estimate)

Description: The total estimated population for the county based on the ACS survey.
Data Type: Integer

Variable Name: DP05_0001M (Total Population Margin of Error)

Description: The margin of error for the total population estimate.
Data Type: Integer

Variable Name: DP05_0018E (Median Age Estimate)

Description: The estimated median age of the population in the county.
Data Type: Float

Variable Name: DP05_0018M (Median Age Margin of Error)

Description: The margin of error for the median age estimate.
Data Type: Float

Variable Name: DP05_0037PE (Percent of Population that is White)

Description: The percentage of the population identified as white.
Data Type: Float

Variable Name: DP05_0037PM (Percent White Margin of Error)

Description: The margin of error for the percent white estimate.
Data Type: Float

Variable Name: DP05_0071PE (Percent of Population that is Hispanic or Latino)

Description: The percentage of the population identified as Hispanic or Latino.
Data Type: Float

Variable Name: DP05_0071PM (Percent Hispanic or Latino Margin of Error)

Description: The margin of error for the percent Hispanic or Latino estimate.
Data Type: Float


In [22]:
# Import necessary libraries
import pandas as pd

In [48]:
# Loading COVID-19 cases, deaths, and population data
covid_cases = pd.read_csv('covid_confirmed_usafacts.csv')
covid_deaths = pd.read_csv('covid_deaths_usafacts.csv')
covid_population = pd.read_csv('covid_county_population_usafacts.csv')

# Show first few rows of datasets
print("COVID-19 Cases:")
display(covid_cases.head())

print("COVID-19 Deaths:")
display(covid_deaths.head())

print("COVID-19 Population:")
display(covid_population.head())


COVID-19 Cases:


countyFIPS            County Name State  StateFIPS  2020-01-22  2020-01-23  \
0           0  Statewide Unallocated    AL          1           0           0   
1        1001        Autauga County     AL          1           0           0   
2        1003        Baldwin County     AL          1           0           0   
3        1005        Barbour County     AL          1           0           0   
4        1007           Bibb County     AL          1           0           0   

   2020-01-24  2020-01-25  2020-01-26  2020-01-27  ...  2023-07-14  \
0           0           0           0           0  ...           0   
1           0           0           0           0  ...       19913   
2           0           0           0           0  ...       70521   
3           0           0           0           0  ...        7582   
4           0           0           0           0  ...        8149   

   2023-07-15  2023-07-16  2023-07-17  2023-07-18  2023-07-19  2023-07-20  \
0           0           0           0           0           0           0   
1       19913       19913       19913       19913       19913       19913   
2       70521       70521       70521       70521       70521       70521   
3        7582        7582        7582        7582        7582        7582   
4        8149        8149        8149        8149        8149        8149   

   2023-07-21  2023-07-22  2023-07-23  
0           0           0           0  
1       19913       19913       19913  
2       70521       70521       70521  
3        7582        7582        7582  
4        8149        8149        8149  

[5 rows x 1269 columns]

COVID-19 Deaths:


countyFIPS            County Name State  StateFIPS  2020-01-22  2020-01-23  \
0           0  Statewide Unallocated    AL          1           0           0   
1        1001        Autauga County     AL          1           0           0   
2        1003        Baldwin County     AL          1           0           0   
3        1005        Barbour County     AL          1           0           0   
4        1007           Bibb County     AL          1           0           0   

   2020-01-24  2020-01-25  2020-01-26  2020-01-27  ...  2023-07-14  \
0           0           0           0           0  ...           0   
1           0           0           0           0  ...         235   
2           0           0           0           0  ...         731   
3           0           0           0           0  ...         104   
4           0           0           0           0  ...         111   

   2023-07-15  2023-07-16  2023-07-17  2023-07-18  2023-07-19  2023-07-20  \
0           0           0           0           0           0           0   
1         235         235         235         235         235         235   
2         731         731         731         731         731         731   
3         104         104         104         104         104         104   
4         111         111         111         111         111         111   

   2023-07-21  2023-07-22  2023-07-23  
0           0           0           0  
1         235         235         235  
2         731         731         731  
3         104         104         104  
4         111         111         111  

[5 rows x 1269 columns]

COVID-19 Population:


countyFIPS            County Name State  population
0           0  Statewide Unallocated    AL           0
1        1001         Autauga County    AL       55869
2        1003         Baldwin County    AL      223234
3        1005         Barbour County    AL       24686
4        1007            Bibb County    AL       22394

In [49]:
# Load ACS Demographic dataset
acs_data = pd.read_csv('ACSDP5Y2020.DP05-Data.csv')

# Show first few rows 
print("ACS Demographic Data:")
display(acs_data.head())



ACS Demographic Data:


GEO_ID                  NAME                               DP05_0001E  \
0  Geography  Geographic Area Name  Estimate!!SEX AND AGE!!Total population   
1  0100000US         United States                                326569308   

                                       DP05_0001M  \
0  Margin of Error!!SEX AND AGE!!Total population   
1                                           *****   

                                      DP05_0002E  \
0  Estimate!!SEX AND AGE!!Total population!!Male   
1                                      160818530   

                                          DP05_0002M  \
0  Margin of Error!!SEX AND AGE!!Total population...   
1                                               7991   

                                        DP05_0003E  \
0  Estimate!!SEX AND AGE!!Total population!!Female   
1                                        165750778   

                                          DP05_0003M  \
0  Margin of Error!!SEX AND AGE!!Total population...   
1                                               8011   

                                          DP05_0004E  \
0  Estimate!!SEX AND AGE!!Total population!!Sex r...   
1                                               97.0   

                                          DP05_0004M  ...  \
0  Margin of Error!!SEX AND AGE!!Total population...  ...   
1                                                0.1  ...   

                                         DP05_0085PM  \
0  Percent Margin of Error!!HISPANIC OR LATINO AN...   
1                                                0.1   

                    DP05_0086PE                                   DP05_0086PM  \
0  Percent!!Total housing units  Percent Margin of Error!!Total housing units   
1                           (X)                                           (X)   

                                         DP05_0087PE  \
0  Percent!!CITIZEN, VOTING AGE POPULATION!!Citiz...   
1                                          233440758   

                                         DP05_0087PM  \
0  Percent Margin of Error!!CITIZEN, VOTING AGE P...   
1                                                (X)   

                                         DP05_0088PE  \
0  Percent!!CITIZEN, VOTING AGE POPULATION!!Citiz...   
1                                               48.5   

                                         DP05_0088PM  \
0  Percent Margin of Error!!CITIZEN, VOTING AGE P...   
1                                                0.1   

                                         DP05_0089PE  \
0  Percent!!CITIZEN, VOTING AGE POPULATION!!Citiz...   
1                                               51.5   

                                         DP05_0089PM Unnamed: 358  
0  Percent Margin of Error!!CITIZEN, VOTING AGE P...          NaN  
1                                                0.1          NaN  

[2 rows x 359 columns]

In [75]:
#The countyFIPS column contains unique codes for each county. 
#This code is converting the countyFIPS values into strings because some county FIPS codes have leading zeros. 
#The .zfill(5) makes sure that each countyFIPS code is 5 characters long by adding zeros if needed.

covid_cases['countyFIPS'] = covid_cases['countyFIPS'].astype(str).str.zfill(5)
covid_deaths['countyFIPS'] = covid_deaths['countyFIPS'].astype(str).str.zfill(5)
covid_population['countyFIPS'] = covid_population['countyFIPS'].astype(str).str.zfill(5)

def extract_fips(geo_id):
    if "US" in geo_id:
        return geo_id.split("US")[1].zfill(5)
    else:
        return None 

# Apply extraction function to ACS dataset
acs_data['countyFIPS'] = acs_data['GEO_ID'].apply(extract_fips)

# Remove the rows where countyFIPS wasnt extracted
acs_data = acs_data.dropna(subset=['countyFIPS'])

# Melt datasets 
covid_cases_melted = covid_cases.melt(id_vars=['countyFIPS', 'County Name', 'State', 'StateFIPS'], 
                                      var_name='Date', value_name='ConfirmedCases')
covid_deaths_melted = covid_deaths.melt(id_vars=['countyFIPS', 'County Name', 'State', 'StateFIPS'], 
                                        var_name='Date', value_name='Deaths')

# Show melted datasets 
print("Melted COVID-19 Cases:")
display(covid_cases_melted.head())

print("Melted COVID-19 Deaths:")
display(covid_deaths_melted.head())



Melted COVID-19 Cases:


countyFIPS            County Name State  StateFIPS        Date  \
0      00000  Statewide Unallocated    AL          1  2020-01-22   
1      01001        Autauga County     AL          1  2020-01-22   
2      01003        Baldwin County     AL          1  2020-01-22   
3      01005        Barbour County     AL          1  2020-01-22   
4      01007           Bibb County     AL          1  2020-01-22   

   ConfirmedCases  
0               0  
1               0  
2               0  
3               0  
4               0

Melted COVID-19 Deaths:


countyFIPS            County Name State  StateFIPS        Date  Deaths
0      00000  Statewide Unallocated    AL          1  2020-01-22       0
1      01001        Autauga County     AL          1  2020-01-22       0
2      01003        Baldwin County     AL          1  2020-01-22       0
3      01005        Barbour County     AL          1  2020-01-22       0
4      01007           Bibb County     AL          1  2020-01-22       0

In [74]:
# Merge the COVID-19 cases and deaths 
covid_combined = covid_cases_melted.merge(covid_deaths_melted, 
                                          on=['countyFIPS', 'County Name', 'State', 'StateFIPS', 'Date'])

# Merge the combined dataset with population data
covid_combined = covid_combined.merge(covid_population, on=['countyFIPS', 'County Name', 'State'])

# Show the first few rows of the combined dataset
print("Combined COVID-19 Data:")
display(covid_combined.head())



Combined COVID-19 Data:


countyFIPS            County Name State  StateFIPS        Date  \
0      00000  Statewide Unallocated    AL          1  2020-01-22   
1      00000  Statewide Unallocated    AL          1  2020-01-23   
2      00000  Statewide Unallocated    AL          1  2020-01-24   
3      00000  Statewide Unallocated    AL          1  2020-01-25   
4      00000  Statewide Unallocated    AL          1  2020-01-26   

   ConfirmedCases  Deaths  population  
0               0       0           0  
1               0       0           0  
2               0       0           0  
3               0       0           0  
4               0       0           0

In [57]:
# Merge the combined COVID-19 data with the ACS data 
merged_data = covid_combined.merge(acs_data, on='countyFIPS', how='inner')

# Show the first few rows of the merged dataset
print("Merged COVID-19 and ACS Data:")
display(merged_data.head())
 

Merged COVID-19 and ACS Data:


countyFIPS            County Name State  StateFIPS        Date  \
0      00000  Statewide Unallocated    AL          1  2020-01-22   
1      00000  Statewide Unallocated    AL          1  2020-01-23   
2      00000  Statewide Unallocated    AL          1  2020-01-24   
3      00000  Statewide Unallocated    AL          1  2020-01-25   
4      00000  Statewide Unallocated    AL          1  2020-01-26   

   ConfirmedCases  Deaths  population     GEO_ID           NAME  ...  \
0               0       0           0  0100000US  United States  ...   
1               0       0           0  0100000US  United States  ...   
2               0       0           0  0100000US  United States  ...   
3               0       0           0  0100000US  United States  ...   
4               0       0           0  0100000US  United States  ...   

  DP05_0085PM DP05_0086PE DP05_0086PM DP05_0087PE DP05_0087PM DP05_0088PE  \
0         0.1         (X)         (X)   233440758         (X)        48.5   
1         0.1         (X)         (X)   233440758         (X)        48.5   
2         0.1         (X)         (X)   233440758         (X)        48.5   
3         0.1         (X)         (X)   233440758         (X)        48.5   
4         0.1         (X)         (X)   233440758         (X)        48.5   

  DP05_0088PM DP05_0089PE DP05_0089PM Unnamed: 358  
0         0.1        51.5         0.1          NaN  
1         0.1        51.5         0.1          NaN  
2         0.1        51.5         0.1          NaN  
3         0.1        51.5         0.1          NaN  
4         0.1        51.5         0.1          NaN  

[5 rows x 367 columns]

In [78]:
chosen_state = 'AL'
state_data = merged_data[merged_data['State'] == chosen_state]

# Get the most recent 7 days of data
last_7_days = state_data['Date'].unique()[-7:]  # Get the last 7 days' dates
print(f"Last 7 days of data for {chosen_state}: {last_7_days}")

# Filter the data for the last 7 days
state_last_week_data = state_data[state_data['Date'].isin(last_7_days)]

# Calculate trends for the last week
cases_trend = state_last_week_data.groupby('Date')['ConfirmedCases'].sum().diff()

# See if cases are increasing, decreasing, or stable
if all(cases_trend > 0):
    trend = "increasing"
elif all(cases_trend < 0):
    trend = "decreasing"
else:
    trend = "stable"

print(f"COVID-19 cases in {chosen_state} for the last week are {trend}.")


Last 7 days of data for AL: ['2023-07-17' '2023-07-18' '2023-07-19' '2023-07-20' '2023-07-21'
 '2023-07-22' '2023-07-23']
COVID-19 cases in AL for the last week are stable.


In [79]:
# Calculate mean, median, and standard deviation (statistical summary)
mean_cases = total_cases_last_week.mean()
median_cases = total_cases_last_week.median()
std_cases = total_cases_last_week.std()

# Display the results
print(f"Mean cases in {chosen_state} over the last week: {mean_cases}")
print(f"Median cases in {chosen_state} over the last week: {median_cases}")
print(f"Standard deviation of cases in {chosen_state} over the last week: {std_cases}")



Mean cases in AL over the last week: 4264.714285714285
Median cases in AL over the last week: 4265.0
Standard deviation of cases in AL over the last week: 0.7559289460184544


The goal of this project was to analyze COVID-19 data for Alabama over the last week of 2020. I used data on confirmed cases, deaths, and population, as well as demographic information from the American Community Survey (ACS).

-I loaded three datasets: COVID-19 confirmed cases, deaths, and population data for U.S. counties.
-I also loaded the ACS demographic data.

-Converted the countyFIPS codes to strings and ensured consistency across datasets to merge them correctly.
-I used the melt() function to reshape the time-series data (COVID-19 cases and deaths) into a long format for easier analysis.

-Merged the COVID-19 datasets (cases, deaths, and population) using countyFIPS as the key.
Then, merged this combined dataset with the ACS demographic data.


-Filtered the merged dataset to focus only on Alabama (labeled as "AL" in the data).

-Extracted the last 7 days of data to analyze trends in COVID-19 confirmed cases.

-Calculated the daily differences in confirmed cases for Alabama over the past 7 days to determine if cases were increasing, decreasing, or stable.

-Calculated the mean, median, and standard deviation of COVID-19 cases for Alabama over the last 7 days to summarize the data.

Findings:
Trend: COVID-19 cases in Alabama over the past 7 days were stable.

Summary Statistics:
Mean cases: 4264.714285714285
Median cases: 4265.0
Standard deviation: 0.7559289460184544